lib/dl_core/dl_core/united_storage_client.py

```diff
@@ -322,6 +322,15 @@ def set_tenant_override(self, tenant: TenantDef) -> None:
                 f"US client folder ID is immutable with auth context {type(self._auth_ctx).__qualname__}"
             )
 
+    def set_dataset_context(self, dataset_id: Optional[str]) -> None:
+        """
+        Set `X-DL-DatasetId` header for requests or delete if None
+        """
+        if dataset_id is not None:
+            self._extra_headers[DLHeadersCommon.DATASET_ID.value] = dataset_id
+        else:
+            self._extra_headers.pop(DLHeadersCommon.DATASET_ID.value, None)
```

Comment:
I think we should adjust the `_ensure_extra_headers` method above, because right now if we were to call `set_tenant_override` after settings dataset context, dataset context would be erased